js/ui/appDisplay.js

```diff
@@ -2743,3 +2780,80 @@ class AppCenterIcon extends AppIcon {
         return true;
     }
 });
+
+const HackAppIconState = {
+    ACTIVATED: ViewIconState.NUM_STATES + 2,
```

Comment:
Where are these values coming from?

Reply:
This comes from: https://github.com/endlessm/gnome-shell/blob/master/js/ui/appDisplay.js#L2660

The `AppCenterIconState` introduces two new states, and what I've done here is to add two more. I can add this as a comment in the code to clarify why +2 and +3.

Reply:
It seems a bit fragile to add values to an existing enum all over the place, what if `AppCenterIconState` gets another value added? But I guess if we do it already for the app center, it's out of scope to fix it for Hack.

To minimize the risk, how about renaming `AppCenterIconState`  to `DownstreamEndlessIconState` and merging these values into there?

Reply:
Done, I've renamed the `AppCenterIconState` to `DownstreamEndlessIconState` and merged there the AppCenter and the HackIcon keys.